File: javascript/2364. Count Number of Bad Pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countBadPairs = function (nums) {
  let ans = 0;
  let map = new Map();
  for (let i = 0; i < nums.length; i++) {
    let val = i - nums[i];
    if (map.has(val)) {
      ans += i - map.get(val);
      map.set(val, map.get(val) + 1);
    } else {
      ans += i;
      map.set(val, 1);
    }
  }

  return ans;
};

File: c#/2364. Count Number of Bad Pairs.cs
public class Solution {
    public long CountBadPairs(int[] nums) {
        Dictionary<int, int> map = new();
        long ans = 0;
        for (int i = 0; i < nums.Length; i++) {
            int val = i - nums[i];

            if (map.ContainsKey(val)) {
                ans += i - map[val];
                map[val]++;
            } 
            else {
                ans += i;
                map[val] = 1;
            }
        }

        return ans;
    }
}